Tests/AttributedStringStyleTests.swift

```diff
@@ -555,6 +555,134 @@ class StringStyleTests: XCTestCase {
         XCTAssertFalse(stillHasAltSixDict)
     }
 
+    func testStyleBlockRules() {
+        let string = "0<one attr1=\"11\">1<two attr2=\"12\">2</two></one>"
+
+        var tagAttr1Value: String!
+        var tagAttr2Value: String!
+
+        let rules: [XMLStyleRule] = [
+            .styleBlock("one") { attributes in
+                guard let val1 = attributes["attr1"], let intVal1 = Int(val1) else {
+                    XCTFail("attr1 not found")
+                    return StringStyle()
+                }
+
+                tagAttr1Value = val1
+                return StringStyle(.baselineOffset(CGFloat(intVal1)))
+            },
+            .styleBlock("two") { attributes in
+                guard let val2 = attributes["attr2"], let intVal2 = Int(val2) else {
+                    XCTFail("attr2 not found")
+                    return StringStyle()
+                }
+
+                tagAttr2Value = val2
+                return StringStyle(.baselineOffset(CGFloat(intVal2)))
+            },
+        ]
+
+        let attributed = string.styled(with: .font(BONFont.fontA), .xmlRules(rules))
+        XCTAssertEqual(attributed.string, "012")
+
+        XCTAssertEqual(tagAttr1Value, "11")
+        XCTAssertEqual(tagAttr2Value, "12")
+
+        let attrs0 = attributed.attributes(at: 0, effectiveRange: nil)
+        let attrs1 = attributed.attributes(at: 1, effectiveRange: nil)
+        let attrs2 = attributed.attributes(at: 2, effectiveRange: nil)
+
+        XCTAssertTrue(attrs0.count == 1)
+        XCTAssertTrue(attrs1.count == 2)
+        XCTAssertTrue(attrs2.count == 2)
+
+        XCTAssertNil(attrs0[.baselineOffset])
+        guard let lineSpacing1 = attrs1[.baselineOffset] as? Int else { XCTFail("lineSpacing1 not found"); return }
```

Comment:
You should be able to mark the test method as `throws` and use `let lineSpacing1 = try XCTUnwrap(attrs1[.baselineOffset] as? Int)` to achieve the same effect, for this line and below.